### PR TITLE
homeassistant: update to version 2021.8.8

### LINKED ISSUE #4458: [Bug] ZHA (zigbee) not working after home assistant core update to 2021.1.5
### Setup
_Package Name:_ Home Assistant Core (spk/homeassistant)
_Package Version:_ 2021.1.5

_NAS Model:_ DS214+
_NAS Architecture:_ armadaxp
_DSM version:_ 6.2.3-25426 Update 3

### Expected behavior
Zigbee integration (zha) works. It worked before on package version 0.118.5 (python 3.7 from synology)

### Actual behavior
Home assistant core 2021.1.5 starts up correctly. But when configuring the zha integration, dependency errors are shown in the homeassistant.log (see below).

### Steps to reproduce
_1._ Install python 3.8 from syno community repository
_2._ Install Home Assistant Core 2021.1.5 from syno community repository, start, onboard
_3._ In the homeassistant gui, add the zha integration, which fails with errors in the log

I don't really understand the dependencies with different kind of native/non-native modules/wheels, etc. But I would be happy to try things out and am also able to build an spk on my own.

#### Package log
```
[add zha integration start]

2021-02-27 10:03:01 INFO (SyncWorker_7) [homeassistant.util.package] Attempting install of bellows==0.21.0
2021-02-27 10:04:08 ERROR (SyncWorker_7) [homeassistant.util.package] Unable to install package bellows==0.21.0: ERROR: Command errored out with exit status 1:
   command: /volume1/@appstore/homeassistant/env/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/volume1/@appstore/homeassistant/tmp/pip-install-72lvopei/pycryptodome/setup.py'"'"'; __file__='"'"'/volume1/@appstore/homeassistant/tmp/pip-install-72lvopei/pycryptodome/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /volume1/@appstore/homeassistant/tmp/pip-wheel-bcqzo5ir
       cwd: /volume1/@appstore/homeassistant/tmp/pip-install-72lvopei/pycryptodome/
  Complete output (311 lines):
  Testing support for clang
  Target does not support clang
  Testing support for gcc
  Target does not support gcc
  Testing support for stdint.h header

[skip many copying/creating lines]

  copying lib/Crypto/Math/_IntegerNative.pyi -> build/lib.linux-armv7l-3.8/Crypto/Math
  copying lib/Crypto/Math/_IntegerBase.pyi -> build/lib.linux-armv7l-3.8/Crypto/Math
  copying lib/Crypto/Math/Primality.pyi -> build/lib.linux-armv7l-3.8/Crypto/Math
  running build_ext
  building 'Crypto.Hash._MD2' extension
  creating build/temp.linux-armv7l-3.8
  creating build/temp.linux-armv7l-3.8/src
  /spksrc/toolchain/syno-armv7-6.1/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -I/spksrc/toolchain/syno-armv7-6.1/work/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/usr/include -I/spksrc/spk/python38/work-armv7-6.1/install//var/packages/python38/target/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -L /spksrc/spk/python38/work-armv7-6.1/install/var/packages/python38/target/lib -I /spksrc/spk/python38/work-armv7-6.1/install/var/packages/python38/target/include -I/spksrc/toolchain/syno-armv7-6.1/work/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/usr/include -I/spksrc/spk/python38/work-armv7-6.1/install//var/packages/python38/target/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -L /spksrc/spk/python38/work-armv7-6.1/install/var/packages/python38/target/lib -I /spksrc/spk/python38/work-armv7-6.1/install/var/packages/python38/target/include -fPIC -DPYCRYPTO_LITTLE_ENDIAN -DSYS_BITS=32 -DLTC_NO_ASM -Isrc/ -I/volume1/@appstore/homeassistant/env/include -I/var/packages/python38/target/include/python3.8 -c src/MD2.c -o build/temp.linux-armv7l-3.8/src/MD2.o
  unable to execute '/spksrc/toolchain/syno-armv7-6.1/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-gcc': No such file or directory
  error: command '/spksrc/toolchain/syno-armv7-6.1/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pycryptodome
ERROR: xbee-helper 0.0.7 requires xbee, which is not installed.
ERROR: pygatt 4.0.5 requires enum-compat, which is not installed.
    ERROR: Command errored out with exit status 1:
     command: /volume1/@appstore/homeassistant/env/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/volume1/@appstore/homeassistant/tmp/pip-install-72lvopei/pycryptodome/setup.py'"'"'; __file__='"'"'/volume1/@appstore/homeassistant/tmp/pip-install-72lvopei/pycryptodome/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /volume1/@appstore/homeassistant/tmp/pip-record-ql4o_dc_/install-record.txt --single-version-externally-managed --compile --install-headers /volume1/@appstore/homeassistant/env/include/site/python3.8/pycryptodome
         cwd: /volume1/@appstore/homeassistant/tmp/pip-install-72lvopei/pycryptodome/
    Complete output (311 lines):
    Testing support for clang
    Target does not support clang
    Testing support for gcc

[skip many copying/creating lines]

    copying lib/Crypto/Math/_IntegerNative.pyi -> build/lib.linux-armv7l-3.8/Crypto/Math
    copying lib/Crypto/Math/_IntegerBase.pyi -> build/lib.linux-armv7l-3.8/Crypto/Math
    copying lib/Crypto/Math/Primality.pyi -> build/lib.linux-armv7l-3.8/Crypto/Math
    running build_ext
    building 'Crypto.Hash._MD2' extension
    creating build/temp.linux-armv7l-3.8
    creating build/temp.linux-armv7l-3.8/src
    /spksrc/toolchain/syno-armv7-6.1/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -I/spksrc/toolchain/syno-armv7-6.1/work/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/usr/include -I/spksrc/spk/python38/work-armv7-6.1/install//var/packages/python38/target/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -L /spksrc/spk/python38/work-armv7-6.1/install/var/packages/python38/target/lib -I /spksrc/spk/python38/work-armv7-6.1/install/var/packages/python38/target/include -I/spksrc/toolchain/syno-armv7-6.1/work/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/usr/include -I/spksrc/spk/python38/work-armv7-6.1/install//var/packages/python38/target/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -L /spksrc/spk/python38/work-armv7-6.1/install/var/packages/python38/target/lib -I /spksrc/spk/python38/work-armv7-6.1/install/var/packages/python38/target/include -fPIC -DPYCRYPTO_LITTLE_ENDIAN -DSYS_BITS=32 -DLTC_NO_ASM -Isrc/ -I/volume1/@appstore/homeassistant/env/include -I/var/packages/python38/target/include/python3.8 -c src/MD2.c -o build/temp.linux-armv7l-3.8/src/MD2.o
    unable to execute '/spksrc/toolchain/syno-armv7-6.1/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-gcc': No such file or directory
    error: command '/spksrc/toolchain/syno-armv7-6.1/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /volume1/@appstore/homeassistant/env/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/volume1/@appstore/homeassistant/tmp/pip-install-72lvopei/pycryptodome/setup.py'"'"'; __file__='"'"'/volume1/@appstore/homeassistant/tmp/pip-install-72lvopei/pycryptodome/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /volume1/@appstore/homeassistant/tmp/pip-record-ql4o_dc_/install-record.txt --single-version-externally-managed --compile --install-headers /volume1/@appstore/homeassistant/env/include/site/python3.8/pycryptodome Check the logs for full command output.
WARNING: You are using pip version 20.0.2; however, version 21.0.1 is available.
You should consider upgrading via the '/volume1/@appstore/homeassistant/env/bin/python -m pip install --upgrade pip' command.
2021-02-27 10:04:08 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/aiohttp/web_protocol.py", line 422, in _handle_request
    resp = await self._request_handler(request)
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/aiohttp/web_app.py", line 499, in _handle
    resp = await handler(request)
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/aiohttp/web_middlewares.py", line 118, in impl
    return await handler(request)
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/components/http/security_filter.py", line 56, in security_filter_middleware
    return await handler(request)
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/components/http/request_context.py", line 18, in request_context_middleware
    return await handler(request)
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/components/http/ban.py", line 72, in ban_middleware
    return await handler(request)
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/components/http/auth.py", line 127, in auth_middleware
    return await handler(request)
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/components/http/view.py", line 129, in handle
    result = await result
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/components/config/config_entries.py", line 136, in post
    return await super().post(request)
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/components/http/data_validator.py", line 60, in wrapper
    result = await method(view, request, *args, **kwargs)
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/helpers/data_entry_flow.py", line 69, in post
    result = await self._flow_mgr.async_init(
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/data_entry_flow.py", line 119, in async_init
    flow = await self.async_create_flow(handler, context=context, data=data)
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/config_entries.py", line 532, in async_create_flow
    await async_process_deps_reqs(self.hass, self._hass_config, integration)
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/setup.py", line 343, in async_process_deps_reqs
    await requirements.async_get_integration_with_requirements(
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/requirements.py", line 76, in async_get_integration_with_requirements
    await async_process_requirements(
  File "/volume1/@appstore/homeassistant/env/lib/python3.8/site-packages/homeassistant/requirements.py", line 133, in async_process_requirements
    raise RequirementsNotFound(name, [req])
homeassistant.requirements.RequirementsNotFound: Requirements for zha not found: ['bellows==0.21.0'].
```

### LINKED ISSUE #4490: [Package Request] - Update Home-Assistant Core from 2021.1.5 to 2021.3.4
* Name:  Home-Assistant Core 
* Description: Open source home automation that puts local control and privacy first
* Website:  https://github.com/home-assistant/core/releases/tag/2021.3.4
* Software documentation: https://www.home-assistant.io/
* Build/Installation documentation: https://www.home-assistant.io/installation/
* Source code:  https://github.com/home-assistant/core/releases/tag/2021.3.4
* License: https://github.com/home-assistant/core/blob/dev/LICENSE.md

Current package is at Version 2021.1.5-13 created by @ymartin59 on 2021-02-24 20:42:39.873587 
Update homeassistant from 2021.1.5 to 2021.3.4

https://synocommunity.com/package/homeassistant

### LINKED ISSUE #4761: 'Home Assistant' is not repairing/installing on DS220j after upgrade to DSM 7.0
### Setup
_Package Name:_ [Home Assistant ](https://synocommunity.com/package/homeassistant)
_Package Version:_ 2021.1.5-13

_NAS Model:_ [DS220j](https://global.download.synology.com/download/Document/Hardware/DataSheet/DiskStation/20-year/DS220j/enu/Synology_DS220j_Data_Sheet_enu.pdf?_ga=2.211685909.1610306812.1605904614-1003422424.1605904614&_gl=1*2o7ql3*_ga*MTAwMzQyMjQyNC4xNjA1OTA0NjE0*_ga_9Q8XT3X20D*MTYwNTk2Njg4NC41LjAuMTYwNTk2Njg4NC4w) 
_NAS Architecture:_ rtd1296 
_DSM version:_ DSM 7.0-41890

### Expected behaviour
After the upgrade from DSM 6.2.4-25556 Update 2 it should continue working on DSM 7.0 as well, ~~especially that there are people having it working on 7.0 beta version~~. 

### Actual behaviour
After the upgrade HA made it to 'Repair' list together with Python packages. Whilst Python packages have been "repaired" just fine, repairing or re-installing HA ends up with **"Home Assistant Core: Invalid file format. Please contact the package developer."** error.  

Same error appeared for 'htop' and 'vim' packages, but I can live without them. 

### Steps to reproduce
_1._ Upgrade existing DSM 6 with HA installed and running to DSM 7.
_2._ Try to 'Repair' or re-install HA.

### PR DESCRIPTION
**Pull request by hgy59**
_Motivation:_  Update homeassistant to include security updates for pyyaml, Jinja2 and urllib3
_Linked issues:_  #4475, closes #4490. closes #4458, #4519, closes #4761

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

new integrations installable from UI (since 2021.1.5-13)
- Adax
- AEMET OpenData
- AirNow
- Ambee
- Antifurto365 iAlarm
- ASUSWRT
- AVM FRITZ!Box Call Monitor
- AVM FRITZ!Box Tools
- AVM FRITZ!SmartHome
- Bosh SHC
- Buienradar
- ClimaCell
- CO2 Signal
- Coinbase
- DSMR Slimme Meter
- Elgato Light
- Enphase Envoy
- Ezwiz (#4475)
- FAA Delays
- Flipr
- Forecast.Solar
- Foscam (#4475)
- Freedompro
- Garages Amsterdam
- Google Maps Travel Time
- Growatt Server
- Habitica
- Hive
- home_plus_control
- Honeywell Lyric
- Honeywell Total Connect Comfort (US)
- Huisbaasje
- Keenetic NDMS2 Router
- KMtronic
- Kostal Plenticore Solar Inverter
- Kraken
- Legrand Home+ Control
- LiteJet
- Litter-Robot
- Lutron Caséta
- Mazda Connected Services
- Met Éireann
- Meteoclimatic
- Modern Forms
- motionEye
- Mullvad VPN
- MusicCast
- mutesync
- mysensors
- Nettigo Air Monitor
- Notifications for Android TV / Fire TV
- NOAA Aurora-Sensor
- Nuki
- NZBGet
- Ondilo ICO
- Pentair ScreenLogic
- Philips TV
- Picnic
- Prosegur Alarm
- Renault
- Rheem EcoNet Products
- Rituals Perfume Genie
- SIA Alarmsystems
- SiteSage Emonitor
- SMA Solar
- SmartTub
- Somfy MyLink
- Subaru
- Switcher
- Syncthing
- System-Bridge
- Total Connect
- Verisure
- Wallbox
- Waze Travel Time
- Yale Smart Living
- YouLess
- Z-Wave JS

Integrations that do not work
- Apple TV (new since 2021.1.5, but still fails with dependency error, depends on PyAV i.e. on ffmpeg...)
- SMS notifications via GSM-modem (did not work before)
- Tasmota (fails now to setup mqtt)
- WiLight (dit not work before)
- Z-Wave (deprecated, but new integration zwave_js is available)

### CODE REVIEW

**Comment by Sweepr** [comment]
Is this PR going to be merged sometime this year?

This PR was created 14 day's ago against 2021.4.6, and 2 day's ago 2021.5.1 was released https://github.com/home-assistant/core/releases/tag/2021.5.1

What's holding it?

**Comment by Forge36** [comment]
@Sweepr 3 failing CI checks. Unfortunately windows docker isn't supported, which leaves me currently unable to investigate.

Any chance you have experience in this area?

**Comment by hgy59** [comment]
Be patiant, I am working on this. One open target is to reduce the package size by download of pure python packages at installation time.
I have seen the new version, but every new version takes some hours of verification and finding indirect dependencies.

**Review by ymartin59 — APPROVED** [review]

**Comment by Sweepr** [comment]
"All checks have passed", any e.t.a. on merge into master so we can pull from the package manager?

File: mk/spksrc.service.use_alternate_tmpdir [hunk 1]
```diff
@@ -21,15 +21,15 @@ service_clean_tmpdir ()
 {
   if [[ -e "${TMPDIR}" ]]; then
     if [ -n "$(ls -A ${TMPDIR})" ]; then
-       echo "Cleanup ${TMPDIR}"   >> "${INST_LOG}"
-       rm -rvf ${TMPDIR}/* >> "${INST_LOG}" 2>&1
```
Comment (ymartin59) [inline mk/spksrc.service.use_alternate_tmpdir:25]: Is there any drawback to send activity logs to INST_LOG, maybe with recent DSM versions?

**Review by ymartin59 — APPROVED** [review]

File: mk/spksrc.service.use_alternate_tmpdir (hunk 1 above)
Comment (hgy59) [reply mk/spksrc.service.use_alternate_tmpdir:25]: Yes, it is not allowed to write to the ${INST_LOG} file anymore (except on DSM5).

**Comment by hgy59** [comment]
Makred as WIP, because armv7 packages do not work until the *.whl files are renamed with correct machine name (armv7l instead of arm).

**Comment by Pitcentr** [comment]
hello, we should expect HA for DSM7 can there be an approximate date?

**Comment by v120fr** [comment]
> Makred as WIP, because armv7 packages do not work until the *.whl files are renamed with correct machine name (armv7l instead of arm).

Thanks.
DS116 with arm:
I rename the file *.whl
make pip install *
same for composant use the file requirement.txt
first same error yarl not found, but cp all /usr/lib.. python/site-packages in the python directory homeassistant and:
i start first time in root perhaps the trouble, i can't start from GUI synology (denied on file), but in putty with root. ALL IS OK.
Now i launch it with --daemon

Thanks for your job.

**Comment by hgy59** [comment]
Finally solved the wheel file naming by renaming the wheels at installation time.

This might even work for armv5 (88f6281) but could not test that due to missing such a device.

The real problem behind the naming of cross compiled wheels is, that crossenv names the 32-bit arm platform "arm" instead of "armv7l" and that pip uses the wheel file name to validate compatibility (wheels still define  Tag: cp38-cp38-linux_arm instead of cp38-cp38-linux_armv7l in the WHEEL file)

A special rename for pycryptodome is needed, as the tag is cp35-abi3-linux_armv7l and not cp38-cp38-linux_armv7l.

By excluding pure python wheels from the package, the spk file size is reduced from > 110MB to < 33MB.


As homeassistant has such a lot of dependen python packages there are still incompatible packages after installation.

When an integration fails to install within homeassistant, this can often be solved by restarting homeassistant (either with the restart in homeassistant or the DSM package center).

**Comment by hgy59** [comment]
TESTERs are welcome, particular for ARMv5 (88f6281) and QorIQ (PPC) archs like DS-213air, DS-213, DS-413j, DS-213+, DS-413 or DS-x12.

**Comment by v120fr** [comment]
Thanks, this version 2021.8.8 is ok for DS116 (armv7), installation and launch ok.

**Comment by Sweepr** [comment]
@hgy59 

Any e.t.a. on when it will be available in the https://packages.synocommunity.com/ Package Center?

**Comment by Lylrok** [comment]
Hello

Always KO for :

NAS Model: DS416j
NAS Architecture: MARVELL Armada 388 88F6828 (Armada38x)
DSM Version: DSM 7.0-41890

Thanks for your help

**Comment by hgy59** [comment]
@Lylrok thanks for the feedback.

**Comment by Lylrok** [comment]
Hello @hgy59 

It's ok for me now
image

I don't know if you made any changes but I managed to reinstall the package and it seems to work again 

thank you so much
how can I buy you a coffee? :)

**Comment by hgy59** [comment]
@Lylrok you can do something for you: do not post your public URL 🤝

**Comment by Sweepr** [comment]
@hgy59 

Just updated on DS216+, works like a charm, thank you for the hard work!

**Comment by Lylrok** [comment]
A little feedback on the use of HA
Everything is ok and works as before
My scripts were there, I didn't need to redo them

A big thank you again

**Comment by TRuHa83** [comment]
hello. after upgrading to DSM 7.0, I was able to repair the installation with no problem, but I lost the configuration. When I access, it asks me for data to create a new installation but I cannot continue, the button does not work. how could i solve it? Thanks a lot.

Screenshot_20220219-135645~2

**Comment by Red7103** [comment]
Any info on update to current HomeAssistant?
There is a lot of plugin which are unuseable on Synology version...
Please make an updet!

**Pull request merged**